src/crud/document.py

```
@@ -162,6 +221,82 @@ async def create_documents(
+
+    # If session is specified, ensure document belongs to that session
+    if session_name is not None:
+        stmt = stmt.where(models.Document.session_name == session_name)
+
+    result = await db.execute(stmt)
+    await db.commit()
+
+    if result.rowcount == 0:
+        raise ResourceNotFoundException(
+            f"Document {document_id} not found or does not belong to the specified collection/session"
+        )
```

Ruff (0.14.5)

Avoid specifying long messages outside the exception class

(TRY003)